### PR TITLE
LPS-188308

### PR DESCRIPTION
**Pull request by icgsbr**
## LPS Associated
- [LPS-188308](https://liferay.atlassian.net/browse/LPS-188308)

### CODE REVIEW

**Review by carolmariaabb — CHANGES_REQUESTED** [review]

**Comment by icgsbr** [comment]
Hi @carolmariaabb thanks for the comments! I'm on it!

File: modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-field-type/src/main/java/com/liferay/dynamic/data/mapping/form/field/type/internal/document/library/DocumentLibraryDDMFormFieldTemplateContextContributor.java [hunk 1]
```diff
@@ -640,8 +656,25 @@ private Map<String, Object> _getUploadParameters(
 			return new HashMap<>();
 		}
 
-		long groupId = GetterUtil.getLong(
-			ddmFormFieldRenderingContext.getProperty("groupId"));
+		long groupId = 0;
+		DDMFormInstance ddmFormInstance = _getDDMFormInstance(
+			ddmFormFieldRenderingContext.getDDMFormInstanceId());
```
[the remark this answers is not in the document]
Comment (icgsbr) [reply modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-field-type/src/main/java/com/liferay/dynamic/data/mapping/form/field/type/internal/document/library/DocumentLibraryDDMFormFieldTemplateContextContributor.java:661]: done

File: modules/apps/data-engine/data-engine-js-components-web/src/main/resources/META-INF/resources/js/utils/evaluation.es.js [hunk 2]
```diff
@@ -141,14 +141,16 @@ const doEvaluate = debounce((fieldName, evaluatorContext, callback) => {
...
+
 	makeFetch({
 		body: convertToFormData({
 			languageId: editingLanguageId,
 			p_auth: Liferay.authToken,
 			portletNamespace,
 			serializedFormContext: JSON.stringify({
 				...evaluatorContext,
-				formId,
+				formId: formId ? formId : pathname.split('/').at(-1),
```
[the remark this answers is not in the document]
Comment (icgsbr) [reply modules/apps/data-engine/data-engine-js-components-web/src/main/resources/META-INF/resources/js/utils/evaluation.es.js]: done

File: modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-field-type/src/main/java/com/liferay/dynamic/data/mapping/form/field/type/internal/document/library/DocumentLibraryDDMFormFieldTemplateContextContributor.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (icgsbr) [reply modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-field-type/src/main/java/com/liferay/dynamic/data/mapping/form/field/type/internal/document/library/DocumentLibraryDDMFormFieldTemplateContextContributor.java]: done

File: modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-field-type/src/main/java/com/liferay/dynamic/data/mapping/form/field/type/internal/document/library/DocumentLibraryDDMFormFieldTemplateContextContributor.java [hunk 3]
```diff
@@ -640,8 +656,25 @@ private Map<String, Object> _getUploadParameters(
...
+		}
+		else if (ddmFormFieldRenderingContext.getDDMFormInstanceId() != 0) {
+			groupId = GetterUtil.getLong(
+				ddmFormFieldRenderingContext.getProperty("groupId"));
+
+		}
+		else {
+			groupId = GetterUtil.getLong(
+				ddmFormField.getProperty("groupId"));
+		}
```
[the remark this answers is not in the document]
Comment (icgsbr) [reply modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-field-type/src/main/java/com/liferay/dynamic/data/mapping/form/field/type/internal/document/library/DocumentLibraryDDMFormFieldTemplateContextContributor.java]: My bad, this was for test purposes, I forgot to erase. Done.

[the remark this answers is not in the document]
Comment (icgsbr) [reply modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-field-type/src/main/java/com/liferay/dynamic/data/mapping/form/field/type/internal/document/library/DocumentLibraryDDMFormFieldTemplateContextContributor.java]: Done. I used the ddmFormField for test only, I forgot to erase it.

**Comment by icgsbr** [comment]
Hi @carolmariaabb ! I've just commited the requested changes

**Comment by icgsbr** [comment]
ci:test:sf

**Comment by icgsbr** [comment]
ci:test:stable

**Comment by icgsbr** [comment]
ci:test:relevant

**Review by carolmariaabb — APPROVED** [review]

**Comment by carolmariaabb** [comment]
@icgsbr please fix the related failures in `DocumentLibraryDDMFormFieldTemplateContextContributorTest`. Thanks!

**Comment by icgsbr** [comment]
Thanks for the comment @carolmariaabb ! I'll work on it!

**Pull request closed without merging**